### PR TITLE
[SPARK-13149][SQL][FOLLOWUP]Make FileStreamSource thread-safe and atomic

### PR DESCRIPTION
**Pull request by zsxwing**
- Make FileStreamSource thread-safe to use FileStreamSource in multiple queries.
- Only update FileStreamSource's status before returning to make it atomic.

### CODE REVIEW

**Comment by zsxwing** [comment]
cc @tdas

**Comment by zsxwing** [comment]
retest this please

**Comment by zsxwing** [comment]
retest this please

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala [hunk 1]
```diff
@@ -35,6 +36,7 @@ import org.apache.spark.util.collection.OpenHashSet
  *
  * TODO Clean up the metadata files periodically
  */
+@ThreadSafe
```
Comment (tdas) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala]: i dont know about this. Could you explain briefly how does this annotation help?

Comment (zsxwing) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala]: It's just a tag saying the class is thread safe. I just realized we don't use it in Spark. I will remove it and add document instead.

**Comment by zsxwing** [comment]
retest this please

**Pull request closed without merging**